nominal/core/_clientsbunch.py

```diff
@@ -40,6 +67,34 @@ def write_nominal_batches(self, auth_header: str, data_source_rid: str, request:
         _path = f"/storage/writer/v1/nominal/{data_source_rid}"
         self._request("POST", self._uri + _path, params={}, headers=_headers, data=request)
 
+    def write_nominal_batches_with_metrics(
+        self,
+        auth_header: str,
+        data_source_rid: str,
+        request: bytes,
+        oldest_timestamp: IntegralNanosecondsUTC,
+        newest_timestamp: IntegralNanosecondsUTC,
+    ) -> RequestMetrics:
+        _headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/x-protobuf",
+            "Authorization": auth_header,
+        }
+        _path = f"/storage/writer/v1/nominal/{data_source_rid}"
+        before_req = time.time_ns()
+
+        self._request("POST", self._uri + _path, params={}, headers=_headers, data=request)
```

Comment:
I guess we can get rid of some of this code if you just call the other impl method?